supernovacontroller/sequential/i3c.py

```diff
@@ -29,6 +29,17 @@ def __init__(self, driver: Supernova, controller: TransferController, notificati
         self.bus_voltage = None
 
         self.controller_init()
+    
+    @staticmethod
+    def __get_error_from_response(response : dict):
+        errors = []
+        if response["usb_result"] != "CMD_SUCCESSFUL":
+            errors.append(response["usb_result"]) 
+        if response["manager_result"] != "I3C_CONTROLLER_MGR_NO_ERROR":
+            errors.append(response["manager_result"]) 
+        if "NO_TRANSFER_ERROR" not in response["driver_result"]:
+            errors.extend(response["driver_result"]) 
+        return errors
```

Comment:
Maybe we can investigate if the SDK always reports errors in those 3 response fields. If that's the case, this function can be a helper used by all protocols and, therefore, re-located in another place and imported by each protocol.

Reply:
As far as I understand, this is the standard API for errors coming from the SDK, so it is definately a possibility